### PR TITLE
Update landing page

### PR DESCRIPTION
**Pull request by mrocklin**
This rips out most of the previous content.  It focuses less on Dask's internal design / intent / architecture / goals, and more on how to use the project.

I think (hope?) that the design also comes across a little more cleanly.

https://github.com/dask/dask/assets/306380/a315e9e2-2996-476c-b0d4-5752f1894e2d

cc @jrbourbeau @scharlottej13

### CODE REVIEW

**Comment by mrocklin** [comment]
I've added a screen recording to the original comment so that people can get a sense for how it feels without building locally.

File: docs/source/index.rst [hunk 1]
```diff
@@ -2,132 +2,304 @@
...
+        source, but tend to be easier to set up and use, safer, cheaper, etc..
+        Here is an incomplete list.
+
+        -  `**Coiled** <https://coiled.io>`_ provides a standalone Dask deployment product that works
+           in AWS and GCP.  Coiled notably employs most of the active Dask
+           maintainers today.
+
+           Learn more at :bdg-link-primary:`Coiled <https://coiled.io>`
+
+        -  `**Saturn Cloud** <https://saturncloud.io>`_ provides Dask as part of their hosted platform
```
Comment (scharlottej13) [inline docs/source/index.rst:281]:
```suggestion

      .. _saturn: https://saturncloud.io
      .. |saturn| replace:: **Saturn Cloud**

        -  |saturn|_ provides Dask as part of their hosted platform
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -2,132 +2,304 @@
...
+        -   **Larger-than-memory** execution for single machines, allowing you
+            to process data that is larger than your available RAM
+        -   **Parallel** execution for faster processing
+        -   **Distributed** computation for terabyte-sized datasets
 
-Complex Algorithms
+        Dask Arrays allow scientists and researchers to perform intuitive and
+        sophisticated operations on large datasets but use the
+        familiar NumPy API and memory model.  One Dask array is simply a
+        collection of numpy arrays on different computers.
```
Comment (scharlottej13) [inline docs/source/index.rst:112]:
```suggestion
        collection of NumPy arrays on different computers.
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -143,7 +315,7 @@ messy situations in everyday problems.
 .. toctree::
    :maxdepth: 1
    :hidden:
-   :caption: Fundamentals
+   :caption: API
```
Comment (scharlottej13) [inline docs/source/index.rst:318]: Maybe "Dask APIs"? Or something else to distinguish from the API reference lower down?

**Review by scharlottej13 — COMMENTED** [review]
Thanks Matt, I like these changes overall (especially fewer words), and it's really nice to have the APIs higher up above the fold.

File: docs/source/index.rst [hunk 4]
```diff
@@ -2,132 +2,304 @@
...
-*  **Scales up**: Runs resiliently on clusters with 1000s of cores
-*  **Scales down**: Trivial to set up and run on a laptop in a single process
-*  **Responsive**: Designed with interactive computing in mind, it provides rapid
-   feedback and diagnostics to aid humans
+.. tab-set::
 
-|
+    .. tab-item:: Tasks
 
-.. figure:: images/dask-overview.svg
```
[the remark this answers is not in the document]
Comment (scharlottej13) [reply docs/source/index.rst:33]: When I was newer to Dask I pretty much equated Dask with Dask DataFrame. Looking at this image helped orient me and answer questions like: how do all these collections fit together? what is "distributed"? why do I need to care about task graphs?

File: docs/source/index.rst (hunk 3 above)
Comment (mrocklin) [reply docs/source/index.rst:324]: Hrm, I agree with the concern (mildly agree at least).  I'm not a fan of this particular solution though.

**Comment by mrocklin** [comment]
OK, I've given this a second pass.  I think it's better than what we have today. I plan to merge tomorrow if there are no further comments.

**Pull request merged**